rfc/rfc011-verifiable-credential.md

```diff
@@ -0,0 +1,215 @@
+
+### 3.1 Supported proofs
+
+The following proof types must be supported:
+
+#### 3.1.1 JsonWebSignature2020
+
+resources:
+
+* https://w3c-ccg.github.io/ld-proofs/ 
+* https://tools.ietf.org/html/rfc7797
+* https://w3c-ccg.github.io/lds-jws2020
+
+The algorithm used to create the signature is as follows:
+
+- take the input Verifiable Credential and remove the `proof` field. Store as `raw_vc`.
+- compute the sha256 of the `raw_vc`, store as `hash_vc = sha256(raw_vc)`.  
+- compose a `raw_proof` json object without the `jws` field:
+
+```json
+{
+  "type": "JsonWebSignature2020",
+  "proofPurpose": "assertionMethod",
+  "verificationMethod": "did:nuts:EgFjg8zqN6eN3oiKtSvmUucao4VF18m2Q9fftAeANTBd#twlH6rB8ArZrknmBRWLXhao3FutZtvOm0hnNhcruenI",
+  "created": "2021-03-09T11:44:56.382202+01:00"
+}
+```
+where the `verificationMethod` must be a valid assertionMethod ID from the DID Document. `created` MUST be a RFC3339 compliant time string.
+`type` and `proofPurpose` MUST be entered as above.
+
+- compute the sha256 of the `raw_proof`, store as `hash_proof = sha256(raw_proof)`.
+- concatenate  `hash_proof` with `raw_proof` and store as `payload`.
+- construct the JWS `header` as `{"alg":"ES256","b64":false,"crit":["b64"]}` where the `alg` SHOULD be replaced with a valid algorithm.
+- construct a challenge by base64 encoding the header and payload and join with a `.`: `challenge = base64_rawurlencode(header) + '.' + base64_rawurlencode(challenge)`  
+- sign the bytes from the previous step with the private key corresponding to the `kid`: `sig = sign(challenge, pkey(kid))`
+- create `jws` as `header + ".." + base64_rawurlencode(sig)`
+- place the result in the `jws` field of the `proof`:
+
+```json
+{
+  "type": "JsonWebSignature2020",
+  "proofPurpose": "assertionMethod",
+  "verificationMethod": "did:nuts:EgFjg8zqN6eN3oiKtSvmUucao4VF18m2Q9fftAeANTBd#twlH6rB8ArZrknmBRWLXhao3FutZtvOm0hnNhcruenI",
+  "created": "2021-03-09T11:44:56.382202+01:00",
+  "jws": "eyJhbGciOiJFUzI1NiIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..ZXlKaGJHY2lPaUpGVXpJMU5pSjkuVHVqZndMVVJwcnUzbjhuZklhODB1M1M0LW9LcWY0WUs5S2hoZEktUkZPSzdlbnZJTTdLN1E5SzBSeHhRSzNIVWJPTUJyLVlZX1g0eW1YR0pXOHF4UkEuN0F4a3lZekNXTElPZ2Q5TlpnR3p2aHd2UzZZQ3FpRTRPX3FwWGVOSEN6X091S1c0TmJsWkJueTBkZVhXT0lXZ3JNczF4OTZlNmtnaGZGYTRNd0J3TlE="
+}
+```
+
+### 3.2 Content-type
+
+All VCs have a content-type equal to `application/vc+json` when published on the network.
+All VCs MUST be handled as normal JSON. All issued VCs MUST add the `https://nuts.nl/credentials/v1` context.
+VCs within Nuts MAY NOT specify more than one additional type next to `VerifiableCredential`.
+
+### 3.3 Updates
+
+VCs are not updatable, an update can be performed by revoking the current and issuing a new VC.
+
+### 3.4 Identifiers
+
+VC identifiers MUST be constructed as `DID#id` where `id` is unique for the given issuer.
+
+### 3.5 VC Example
+
+Below is an example of a credential.
```

Comment:
maybe mention that issuer and credential subject (holder) are the same it the example (which can occur), but they don't need to.